ExtraDry/ExtraDry.UploadTools/UploadTools.cs

```diff
@@ -0,0 +1,204 @@
+﻿using ExtraDry.Core;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace ExtraDry.UploadTools
+{
+    public static class UploadTools {
+        /// <summary>
+        /// File extensions that will be rejected regardless of the whitelist settings
+        /// </summary>
+        private readonly static List<string> FileTypeBlacklist = new List<string>();
```

Comment:
Overall, this PR has a *lot* of static methods.  Should these be instances and allow the declaration in DI to create a singleton if that is the desire.  I.e. why force this on a consumer?

Reply:
Statics are just about all gone now. The only real remaining one is the cleaning of the file name, because it ultimately just does some regex and makes sense to be static.